LICENSE

```diff
@@ -0,0 +1,216 @@
+      Work (including but not limited to damages for loss of goodwill,
+      work stoppage, computer failure or malfunction, or any and all
+      other commercial damages or losses), even if such Contributor
+      has been advised of the possibility of such damages.
+
+   9. Accepting Warranty or Additional Liability. While redistributing
+      the Work or Derivative Works thereof, You may choose to offer,
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+
+================================================================
+APACHE IOTDB SUBCOMPONENTS:
+
+The following classes are modified from Apache Hive, which is under Apache License 2.0:
+
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/AstErrorNode.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/AstNode.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/AstNodeOrigin.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/Node.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseDriver.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseError.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseException.java
+iotdb/src/main/java/org/apache/iotdb/db/sql/parse/ParseUtils.java
```

Comment:
these paths are not correct for instance teh first file should be ./server/src/main/java/org/apache/iotdb/db/sql/parse/AstErrorNode.java